File: trabalhoChatORB/TestesdoChatGPT/Chat_Client.py
import sys
from omniORB import CORBA

orb = CORBA.ORB_init(sys.argv, CORBA.ORB_ID)
ior = input("Enter the Chat Server's object reference: ")
obj = orb.string_to_object(ior)
chat = obj._narrow(ChatApp.Chat)

if chat is None:
    raise Exception("Object reference is not a Chat")

username = input("Enter your username: ")
chat.joinChat(username)

while True:
    print("1. Send a message")
    print("2. List users")
    print("3. Leave chat")
    choice = input("Choose an option (1/2/3): ")

    if choice == '1':
        message = input("Enter your message (or 'exit' to quit): ")
        if message.lower() == 'exit':
            break
        chat.sendMessage(username, message)

        received_message = chat.receiveMessage()
        print("Received:", received_message)

    elif choice == '2':
        users = chat.listUsers()
        print("Users online:", users)

    elif choice == '3':
        chat.leaveChat(username)
        break

orb.destroy()

File: trabalhoChatORB/C++/ChatClient.cpp
#include "Chat.hh"
#include <iostream>

int main(int argc, char** argv) {
    try {
        CORBA::ORB_var orb = CORBA::ORB_init(argc, argv);

        CORBA::Object_var obj = orb->string_to_object("corbaname::localhost:1050#ChatServer");
        Chat::ChatServer_var chatServerRef = Chat::ChatServer::_narrow(obj);

        if (CORBA::is_nil(chatServerRef)) {
            std::cerr << "Error: Nil reference." << std::endl;
            return 1;
        }

        chatServerRef->sendMessage("User1", "Hello, Server!");

        char* response = chatServerRef->receiveMessage();
        std::cout << "Server response: " << response << std::endl;

        CORBA::string_free(response);
        orb->destroy();
    } catch (const CORBA::Exception& e) {
        std::cerr << "Exception: " << e << std::endl;
    }

    return 0;
}

